File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<char> helperStack;
        int ans = 0;
        for(char it : s){
            if(it == '('){
                helperStack.push(it);
            }else if(it == ')'){
                int temp = helperStack.size();
                ans = max(ans, temp);
                helperStack.pop();

            }
        }
        return ans;
    }
};

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        helper_stack = []
        ans = 0
        for char in s:
            if char == '(':
                helper_stack.append(char)
            elif char == ')':
                ans = max(ans, len(helper_stack))
                helper_stack.pop()
        return ans